Delete temporary name list file in _gnr_resolve()

lyttonhao opened the pull request:
The temporary file should be removed.

sckott commented:
@lyttonhao thanks. Please add a test in `tests` directory to this PR testing that the file is removed

lyttonhao commented:
Already added.

panks commented:
@lyttonhao I believe in your test you are trying to invoke GNR resolver with POST where it sends a file. You don't need to keep the size of file to 1010 names, as anyway **gnr_resolve()** will slice the list into smaller sizes (<=1000) and then pass it to **_gnr_resolve()**. Anything with size > 300 would go via POST protocol using a file. So you can reduce the size of file containing name (to 301 names), otherwise it would take unnecessary time during test.

panks commented:
@sckott I have replaced xrange with range in gnr, in my current PR. If you restart the build on CI after that, this PR will pass for python3 as well.

lyttonhao commented:
@panks Thanks for your advice. I have changed the query number to 301 in test_gnr_resolve_remove_temporary_file(), and added a new test for testing when queried number is larger than 1000. So I think the large species names file is still needed.

sckott commented:
I've restarted the build...

sckott commented:
thanks @lyttonhao , but your test failed on python 3.4 https://travis-ci.org/sckott/pytaxize/jobs/56682501

Did you test on python 3?

lyttonhao commented:
@sckott , I'm sorry I don't test on python 3.4. I will have a look at it soon.

panks commented on pytaxize/gnr.py:
```diff
@@ -122,6 +123,7 @@ def _gnr_resolve(names='Homo sapiens', source=None, format='json', resolve_once=
+            os.remove('__names_list.txt')
 
     data = []
     for each_result in result_json['data']:
```
This file can be ignored now, as I already took care of file deletion in https://github.com/sckott/pytaxize/pull/30
Just add the test cases.
